app/src/main/java/fr/free/nrw/commons/depictions/GridViewAdapter.java

```diff
@@ -0,0 +1,116 @@
+package fr.free.nrw.commons.depictions;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+
+import com.facebook.drawee.view.SimpleDraweeView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.R;
+
+public class GridViewAdapter extends ArrayAdapter {
+
+        private List<Media> data;
+
+        public GridViewAdapter(Context context, int layoutResourceId, List<Media> data) {
+            super(context, layoutResourceId, data);
+            this.data = data;
+        }
+
+        /**
+         * Adds more item to the list
+         * Its triggered on scrolling down in the list
+         * @param images
+         */
+        public void addItems(List<Media> images) {
+            if (data == null) {
+                data = new ArrayList<>();
+            }
+            data.addAll(images);
+            notifyDataSetChanged();
+        }
+
+        /**
+         * Check the first item in the new list with old list and returns true if they are same
+         * Its triggered on successful response of the fetch images API.
+         * @param images
+         */
+        public boolean containsAll(List<Media> images){
+            if (images == null || images.isEmpty()) {
+                return false;
+            }
+            if (data == null) {
+                data = new ArrayList<>();
+                return false;
+            }
+            if (data.size() <= 0) {
```

Comment:
Size will never be negative right , shall we not add an empty check instead ?

Reply:
replaced with `if (data.isEmpty())` in the next commit